controllers/articles.js

```diff
@@ -267,4 +281,137 @@ export default class ArticleController {
       });
     }
   }
+
+  /**
+ * @description - Get all articles
+ * @static
+ * @param {Object} req - the request object
+ * @param {Object} res - the response object
+ * @memberof ArticleController
+ * @returns {Object} class instance
+ */
+  static async getAllArticles(req, res) {
+    try {
+      let {
+        query: {
+          offset, limit
+        }
+      } = req;
+      offset = Number(offset) || 0;
+      limit = Number(limit) || 10;
+      const results = await Article.findAndCountAll({
+        where: {},
+        offset,
+        limit,
+        include: [
+          {
+            model: User,
+            as: 'author',
+            attributes: ['username', 'bio', 'name']
+          },
+          {
+            model: Category,
+            as: 'category',
+            attributes: ['categoryName']
+          },
+          {
+            model: Ratings,
+          }
+        ]
+      });
+      const { count } = results;
+      const meta = Paginate({ count, limit, offset });
+      return res.status(200).json({
+        success: true,
+        results,
+        ...meta
+      });
+    } catch (error) {
+      return res.status(500).json({
+        succes: false,
+        errors: ['Oops, something wrong occured.']
+      });
+    }
+  }
+
+  /**
+  * @description - Get a specific number of articles with criteria
+  * @static
+  * @param {Object} req - the request object
+  * @param {Object} res - the response object
+  * @memberof ArticleController
+  * @returns {Object} class instance
+  */
+  static async getArticlesByHighestField(req, res) {
+    try {
+      const {
+        query: {
+          amount, type
+        }
+      } = req;
+      let articles;
+      if (type === 'latest') {
+        articles = await Article.findAll({
+          where: {},
+          limit: Number(amount) || 5,
+          order: [['createdAt', 'ASC']],
+          include: [
+            {
+              model: User,
+              as: 'author',
+              attributes: ['username', 'bio', 'name']
+            },
+          ]
+        });
+      } else if (type === 'ratings') {
+        articles = await Article.findAll({
+          where: {},
+          order: [[Ratings, 'rating', 'ASC']],
+          limit: Number(amount) || 5,
+          include: [
+            {
+              model: User,
+              as: 'author',
+              attributes: ['username', 'bio', 'name']
+            },
+            {
+              model: Comment
+            },
+            {
+              model: Ratings,
+            },
+          ],
+        });
+      } else {
+        articles = await Article.findAll({
+          where: {},
+          order: [[Comment, 'comment', 'ASC']],
```

Comment:
@leksyib have you tested this to ensure it actually returns the result in the right order?

Reply:
Yes I have @9jaswag

Reply:
@leksyib  I'm surprised that `ASC` returns the right data. What does `DESC` return?

Reply:
`DESC` returns the data with the lowest below the highest. So the ones with the highest ratings come first and the lowest will come after